floris/tools/optimization/layout_optimization/layout_optimization_pyoptsparse.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2022 NREL
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# See https://floris.readthedocs.io for documentation
+
+
+import numpy as np
+import matplotlib.pyplot as plt
+from shapely.geometry import Point
+from scipy.spatial.distance import cdist
+
+from .layout_optimization_base import LayoutOptimization
+
+class LayoutOptimizationPyOptSparse(LayoutOptimization):
+    def __init__(
+        self,
+        fi,
+        boundaries,
+        min_dist=None,
+        freq=None,
+        solver=None,
+        optOptions=None,
+    ):
+        super().__init__(fi, boundaries, min_dist=min_dist, freq=freq)
+        self._reinitialize(solver=solver, optOptions=optOptions)
+
+    def _reinitialize(self, solver=None, optOptions=None):
+        try:
+            import pyoptsparse
+        except ImportError:
+            err_msg = (
+                "It appears you do not have pyOptSparse installed. "
+                + "Please refer to https://pyoptsparse.readthedocs.io/ for "
+                + "guidance on how to properly install the module."
+            )
+            self.logger.error(err_msg, stack_info=True)
+            raise ImportError(err_msg)
+
+        # Insantiate ptOptSparse optimization object with name and objective function
+        self.optProb = pyoptsparse.Optimization('layout', self._obj_func)
+
+        self.optProb = self.add_var_group(self.optProb)
+        self.optProb = self.add_con_group(self.optProb)
+        self.optProb.addObj("obj")
+
+        if solver is not None:
```

Comment:
The way it's defined now, the following situation can occur:
1. You initialize the optimization class with `solver="Nelder-Mead"` and certain `optOptions`.
2. You want to try another set of optimization settings, `newOptions`, so you do `opt._reinitialize(optOptions=newOptions)`.
3. With (2), you have unknowingly overwritten the solver to `SLSQP`.

To prevent this, I would just rename `_reinitialize` to `_initialize`, and assume you only call this function once -- when the class is created. If the user wants to try different settings, they just recreate the optimization class from scratch with `opt = LayoutOptimizationPyOptSparse(...)`.

Reply:
Ah, good catch! I have just removed the `_reinitialize` class altogether and merged the required functions with the `__init__` method.